### PR TITLE
Couple minor GNSS fixes

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Renaming a binding, fixing some prints.

### CODE REVIEW

File: drivers/gnss/gnss_dump.c [hunk 1]
```diff
@@ -94,20 +95,15 @@ int gnss_dump_info(char *str, uint16_t strsize, const struct gnss_info *info)
...
-		       nav_data->latitude % 1000000000, nav_data->longitude / 1000000000,
-		       nav_data->longitude % 1000000000, nav_data->bearing / 1000,
-		       nav_data->bearing % 1000, nav_data->speed / 1000, nav_data->speed % 1000,
-		       altitude_int, altitude_fraction);
+	ret = snprintk(str, strsize, fmt,
+		       nav_data->latitude / 1000000000, llabs(nav_data->latitude % 1000000000),
+		       nav_data->longitude / 1000000000, llabs(nav_data->longitude % 1000000000),
+		       nav_data->bearing / 1000, nav_data->bearing % 1000,
+		       nav_data->speed / 1000, nav_data->speed % 1000,
+		       nav_data->altitude / 1000, abs(nav_data->altitude % 1000));
```
Comment (fabiobaltieri) [inline drivers/gnss/gnss_dump.c:106]: ended up refactoring this as well, lmk what you think

File: drivers/gnss/gnss_dump.c [hunk 2]
```diff
@@ -94,20 +95,15 @@ int gnss_dump_info(char *str, uint16_t strsize, const struct gnss_info *info)
 int gnss_dump_nav_data(char *str, uint16_t strsize, const struct navigation_data *nav_data)
 {
 	int ret;
-	int32_t altitude_int;
-	int32_t altitude_fraction;
 	const char *fmt = "navigation_data: {latitude: %lli.%lli, longitude : %lli.%lli, "
```
[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply drivers/gnss/gnss_dump.c]: Oh, yeah you are right! Ok I fixed that as well. I'll put together a test as well, I have one from input that should be mostly copypastable.

[the remark this answers is not in the document]
Comment (fabiobaltieri) [reply drivers/gnss/gnss_dump.c]: oof, negative values <1 are also off

**Comment by fabiobaltieri** [comment]
- fixed leading 0 and small negative numbers
- moved the publish header in `includes/`
- added a testcase for the dump functionalities

**Review by bjarki-andreasen — DISMISSED** [review]
Very nice work!

**Review by cfriedt — DISMISSED** [review]

**Review by cfriedt — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Pull request merged**